### PR TITLE
Nyxt bundled dependencies

### PR DESCRIPTION
**Pull request by Uthar**
## Description of changes

I've switched it to building with make from the released tarball. It's been recommended by André. Link to [thread](https://github.com/NixOS/nixpkgs/issues/302795).

I've removed mailcap and enchant. I couldn't find any references to mailcap in sources. It seems to be using enchant1 while nixpkgs has enchant2.

I've added xdg-open to the wrapped path. It's needed to open some links.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by aadcg** [comment]
@Uthar this is the way to go! Unfortunately, I'm not a Nix user and can't provide much insight. Thanks for working on it. I hope this PR will be merged :)

**Comment by twitchy0** [comment]
I'm able to execute remote commands on nyxt when building with these changes.

**Review by Vonfry — APPROVED** [review]
Test the binary with some basic functions.

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 321845` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nyxt</li>
  </ul>
</details>

**Comment by tomberek** [comment]
```
nyxt> ; file: /build/source/_build/slime/swank/sbcl.lisp
nyxt> ; in: DEFUN SWANK/SBCL::MAKE-SOCKET-IO-STREAM
nyxt> ;     (EQ :FD-HANDLER SWANK:*COMMUNICATION-STYLE*)
nyxt> ;
nyxt> ; caught WARNING:
nyxt> ;   undefined variable: SWANK:*COMMUNICATION-STYLE*
nyxt>
nyxt>
nyxt> ; file: /build/source/_build/slime/swank/backend.lisp
nyxt> ; in: SWANK/BACKEND::DEFINTERFACE SWANK/BACKEND:WAIT-FOR-INPUT
nyxt> ;     (ERROR
nyxt> ;      "~s not implemented. Check if ~s = ~s is supported by the implementation."
nyxt> ;      'SWANK/BACKEND:WAIT-FOR-INPUT 'SWANK:*COMMUNICATION-STYLE*
nyxt> ;      SWANK:*COMMUNICATION-STYLE*)
nyxt> ;
nyxt> ; caught WARNING:
nyxt> ;   undefined variable: SWANK:*COMMUNICATION-STYLE*
nyxt>
nyxt>
nyxt> ; file: /build/source/_build/s-xml/src/xml.lisp
nyxt> ; in: DEFUN S-XML:PARSE-ATTRIBUTE-VALUE
nyxt> ;     (IF S-XML:*IGNORE-NAMESPACES*
nyxt> ;         (COPY-SEQ STRING)
nyxt> ;         STRING)
nyxt> ;
nyxt> ; caught WARNING:
nyxt> ;   undefined variable: S-XML:*IGNORE-NAMESPACES*
nyxt>
nyxt>
nyxt> ; file: /build/source/_build/parenscript/src/utils.lisp
nyxt> ; in: DEFUN PARENSCRIPT::JS-TARGET-AT-LEAST
nyxt> ;     (PARENSCRIPT::PARSE-SEMVER PARENSCRIPT:*JS-TARGET-VERSION*)
nyxt> ;
nyxt> ; caught WARNING:
nyxt> ;   undefined variable: PARENSCRIPT:*JS-TARGET-VERSION*
nyxt>
nyxt>
nyxt> ; file: /build/source/_build/sly/contrib/slynk-stickers.lisp
```

**Comment by Uthar** [comment]
@ofborg build nyxt

**Comment by abueide** [comment]
is this able to be merged?

**Comment by Uthar** [comment]
@ofborg build nyxt

There's a problem with sbcl 2.4.7 during gc - I pinned it to the previous release.

**Review by mjoerg — COMMENTED** [review]
Version 3.11.8 is available.

**Review by mjoerg — COMMENTED** [review]
Result of `nixpkgs-review pr 321845` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nyxt</li>
  </ul>
</details>

File: pkgs/applications/networking/browsers/nyxt/default.nix [hunk 1]
```diff
@@ -1,56 +1,104 @@
...
-, glib, gdk-pixbuf, cairo
-, mailcap, pango, gtk3
-, glib-networking, gsettings-desktop-schemas
-, xclip, wl-clipboard, notify-osd, enchant
+{ stdenv
+, lib
+, testers
+, wrapGAppsHook3
+, fetchzip
+, sbcl_2_4_6
```
Comment (hraban) [inline pkgs/applications/networking/browsers/nyxt/default.nix:6]: I think the convention for other packages is to just accept the bare program name, and pin it explicitly in all-packages.nix, though I may be mistaking

**Review by 0poss — APPROVED** [review]

**Pull request merged**